code/modules/projectiles/projectile/energy.dm

```diff
@@ -147,3 +147,63 @@
 	armour_penetration = 10 // It can have a little armor pen, as a treat. Bigger than it looks, energy armor is often low.
 	shield_buster = TRUE
 	is_reflectable = FALSE //I will let eswords block it like a normal projectile, but it's not getting reflected, and eshields will take the hit hard.
+
+/obj/item/projectile/energy/detective
+	name = "placeholder name"
+	icon_state = "omnilaser"
+	light_color = LIGHT_COLOR_CYAN
+	damage = 5
+	stamina = 25
+	eyeblur = 5
+
+/obj/item/projectile/energy/detective/overcharged
+	name = "overcharged shot"
+	icon_state = "spark"
+	light_color = LIGHT_COLOR_DARKRED
+	damage = 45
+	stamina = 15
+	eyeblur = 10
+
+/obj/item/projectile/energy/detective/tracker_warrant_shot
+	name = "tracker shot"
+	icon_state = "purple_laser"
+	light_color = LIGHT_COLOR_PURPLE
+	stamina = 0
+
+/obj/item/projectile/energy/detective/tracker_warrant_shot/on_hit(atom/target)
+	. = ..()
+	if(!ishuman(target))
+		no_worky(target)
+		return
+	start_tracking(target)
+	set_warrant(target)
+
+/obj/item/projectile/energy/detective/tracker_warrant_shot/proc/start_tracking(atom/target)
+	for(var/obj/item/gun/energy/detective/D in firer)
+		if(D.tracking_target_UID)
```

Comment:
what if they drop the gun? might be a better idea to use a reference to the firearm on the projectile.